src/components/structures/FavouriteMessagesView/FavouriteMessagesView.tsx

```diff
@@ -0,0 +1,134 @@
+
+    registerFavouritesChangedListener(() => recalcEvents());
+
+    const handleSearchQuery = (query: string) => {
+        setSearchQuery(query);
+    };
+
+    const props = {
+        favouriteMessageEvents,
+        resizeNotifier,
+        searchQuery,
+        handleSearchQuery,
+        cli: matrixClient,
+    };
+
+    return <FavouriteMessagesPanel {...props} />;
+};
+
+function filterFavourites(searchQuery: string, favouriteMessages: FavouriteStorage[]): FavouriteStorage[] {
+    return favouriteMessages.filter((f) =>
+        f.content.body.trim().toLowerCase().includes(searchQuery.trim().toLowerCase()),
+    );
+}
+
+/** If the event was edited, update it with the replacement content */
+async function updateEventIfEdited(event: MatrixEvent, matrixClient: MatrixClient) {
+    const roomId = event.getRoomId();
+    const eventId = event.getId();
+    if (roomId && eventId) {
+        const { events } = await matrixClient.relations(roomId, eventId, RelationType.Replace, null, { limit: 1 });
+        const editEvent = events?.length > 0 ? events[0] : null;
+        if (editEvent) {
+            event.makeReplaced(editEvent);
+        }
+    }
+}
+
+/**
+ * Use the supplied MatrixClient to fetch the event specified in favourite.
+ * Takes a RunState and gives up early if runState.isCancelled().
+ */
+async function fetchEvent(favourite: FavouriteStorage, matrixClient: MatrixClient): Promise<MatrixEvent | null> {
+    try {
+        const evJson = await matrixClient.fetchRoomEvent(favourite.roomId, favourite.eventId);
+        const event = new MatrixEvent(evJson);
+        const roomId = event?.getRoomId();
+        const room = roomId ? matrixClient.getRoom(roomId) : null;
+        if (!event || !room) {
+            return null;
+        }
+
+        // Decrypt the event
+        if (event.isEncrypted()) {
+            // Modifies the event in-place (!)
+            await matrixClient.decryptEventIfNeeded(event);
+        }
+
+        // Inject sender information
+        const sender = event.getSender();
+        if (sender) {
+            event.sender = room.getMember(sender)!;
+        }
+
+        await updateEventIfEdited(event, matrixClient);
+
+        return event;
+    } catch (err) {
+        logger.error(err);
+        return null;
+    }
+}
+
+/**
+ * Use the supplied MatrixClient to fetch all the events for the supplies
+ * favouriteMessages, filtered using searchQuery.
+ * Takes a RunState and gives up early if runState.isCancelled().
+ */
+async function calcEvents(
+    searchQuery: string,
+    favouriteMessages: FavouriteStorage[],
+    matrixClient: MatrixClient,
+): Promise<MatrixEvent[]> {
+    const displayedFavourites: FavouriteStorage[] = filterFavourites(searchQuery, favouriteMessages);
+    const promises: Promise<MatrixEvent | null>[] = displayedFavourites.map((f) => fetchEvent(f, matrixClient));
+    const events = await Promise.all(promises);
+    return events.filter((e) => e !== null);
```

Comment:
```suggestion
    return events.filter((e) => e !== null) as MatrixEvent[]; // force cast because typescript doesn't understand what `filter` does
```